src/painter.rs

```diff
@@ -234,6 +47,30 @@ pub struct Painter {
     pub screen_rect: Rect,
 }
 
+macro_rules! get_error {
```

Comment:
Might be better to name this macro as get_gl_error as that makes the intent even more clear.

Reply:
I decided to rebase my pull request as I found an issue on how the shader sources were being propagated to outside the crate. The rebasing also addresses your previous comment.